### ISSUE TITLE
Simplify instance form input handling

### ISSUE THREAD

**Issue by andraskvr** [inmanta/web-console#193]
There have been multiple tickets about issues with creating and editing instances with various types. While in the current situation (after #140) it's correct (for the most part), the code that handles it has become more and more complicated with the fixes. 
We should take a step back and simplify it, before we can support handling the embedded entities properly.

Some important guidelines:
- updates to the instance can be done with the `PATCH` method, where only the attributes that were changed should be included (until inmanta/inmanta-lsm#449 is implemented at least)
- the data sent to the server should be type correct (as far as typing in JSON goes at least)
- full type validation based on the openapi definiton of an entity is part of another ticket (#92)
- an optional attribute has its type marked with a `?`, these have to supported as well
- don't forget the default values
- the Patternfly components should be used (https://www.patternfly.org/v4/components/form/design-guidelines#data-inputs) to get the input from the user, even after changing the implementation of the handling of this data
